src/dispatch/__tests__/DataPlaneClient.test.ts

```diff
@@ -132,4 +132,107 @@ describe('DataPlaneClient tests', () => {
             'd37eb756444ebf6f785233714d6d942f2b20f69292fb09533f6b69556eb0ff2b'
         );
     });
+
+    test('when the endpoint contains a path then the fetch reqest url contains the path prefix', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
+            fetchRequestHandler: new FetchHttpHandler(),
+            beaconRequestHandler: new BeaconHttpHandler(),
+            endpoint,
+            region: Utils.AWS_RUM_REGION,
+            credentials: Utils.createAwsCredentials()
+        });
+
+        // Run
+        await client.sendFetch(Utils.PUT_RUM_EVENTS_REQUEST);
+
+        // Assert
+        const signedRequest: HttpRequest = (fetchHandler.mock
+            .calls[0] as any)[0];
+        expect(signedRequest.hostname).toEqual(Utils.AWS_RUM_ENDPOINT.hostname);
+        expect(signedRequest.path).toEqual(
+            `${endpoint.pathname}/appmonitors/application123/`
+        );
+    });
+
+    test('when the endpoint path contains a trailing slash then the fetch reqest url drops the trailing slash', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod/'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
+            fetchRequestHandler: new FetchHttpHandler(),
+            beaconRequestHandler: new BeaconHttpHandler(),
+            endpoint,
+            region: Utils.AWS_RUM_REGION,
+            credentials: Utils.createAwsCredentials()
+        });
+
+        // Run
+        await client.sendFetch(Utils.PUT_RUM_EVENTS_REQUEST);
+
+        // Assert
+        const signedRequest: HttpRequest = (fetchHandler.mock
+            .calls[0] as any)[0];
+        expect(signedRequest.hostname).toEqual(Utils.AWS_RUM_ENDPOINT.hostname);
+        expect(signedRequest.path).toEqual(
+            `${endpoint.pathname.replace(
+                /\/$/,
+                ''
+            )}/appmonitors/application123/`
+        );
+    });
+
+    test('when the endpoint contains a path then the beacon reqest url contains the path prefix', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
```

Comment:
nit: For all the cases we are creating a `DataPlaneClient`, we can extract this into a method called `createDPClient` at the start of the file, and reuse in test cases - this would reduce the file length.

Reply:
There is a tradeoff here. Verbosity gives us clarity into what the test's inputs are. While we could extract method for each instantiation, we would need to navigate to the instantiation to see the configuration for the test.

For test suites, file length is not a problem since each test is a self-contained specification of the program's behavior. (However, having long test methods IS a code smell.)